### PR TITLE
Always run with LSP pull diagnostics in code-spaces.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

File: src/VisualStudio/Core/Def/Implementation/Diagnostics/VisualStudioDiagnosticModeService.cs [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+                    return DiagnosticMode.Pull;
+
+                // Otherwise, defer to the workspace+option to determine what mode we're in.
+                return _workspace.Options.GetOption(option);
+            }
+
+            private bool IsInCodeSpacesServer()
+            {
+                // hack until there is an officially supported free-threaded synchronous platform API to ask this question.
+                return Environment.GetEnvironmentVariable("VisualStudioServerMode") == "1";
```
Comment (CyrusNajmabadi) [inline src/VisualStudio/Core/Def/Implementation/Diagnostics/VisualStudioDiagnosticModeService.cs:75]: this was the actual recommendation from teh team on how to determine if we're on the codespaces server.

File: src/VisualStudio/Core/Def/Implementation/LanguageClient/AbstractInProcLanguageClient.cs [hunk 2]
```diff
@@ -85,9 +86,9 @@ internal abstract class AbstractInProcLanguageClient : ILanguageClient
         /// <summary>
         /// Can be overridden by subclasses to control what capabilities this language client has.
         /// </summary>
-        protected internal abstract VSServerCapabilities GetCapabilities();
+        protected internal abstract Task<VSServerCapabilities> GetCapabilitiesAsync(CancellationToken cancellationToken);
```
Comment (davidwengier) [inline src/VisualStudio/Core/Def/Implementation/LanguageClient/AbstractInProcLanguageClient.cs:89]: Why make this async? Future proofing?

Comment (CyrusNajmabadi) [reply src/VisualStudio/Core/Def/Implementation/LanguageClient/AbstractInProcLanguageClient.cs:89]: i can revert.  i needed it in something experiemntal i was doing.  but didn't need it at the end

Comment (dibarbet) [inline src/VisualStudio/Core/Def/Implementation/LanguageClient/AbstractInProcLanguageClient.cs:89]: does this need to be a task?  I looked through and couldn't find any implementation that was actually async

**Review by dibarbet — APPROVED** [review]

File: src/VisualStudio/Core/Def/Implementation/LanguageClient/InProcLanguageServer.cs [hunk 3]
```diff
@@ -578,8 +578,8 @@ private async Task SendDiagnosticsNotificationAsync(Uri uri, ImmutableArray<LSP.
             var option = document.IsRazorDocument()
                 ? InternalDiagnosticsOptions.RazorDiagnosticMode
                 : InternalDiagnosticsOptions.NormalDiagnosticMode;
-            var diagnostics = diagnosticService.GetPushDiagnostics(document.Project.Solution.Workspace, document.Project.Id, document.Id, id: null, includeSuppressedDiagnostics: false, option, cancellationToken)
-                                               .WhereAsArray(IncludeDiagnostic);
+            var pushDiagnostics = diagnosticService.GetPushDiagnostics(document.Project.Solution.Workspace, document.Project.Id, document.Id, id: null, includeSuppressedDiagnostics: false, option, cancellationToken);
```
[the remark this answers is not in the document]
Comment (CyrusNajmabadi) [reply src/VisualStudio/Core/Def/Implementation/LanguageClient/InProcLanguageServer.cs:581]: yup.

File: src/VisualStudio/Core/Def/Implementation/LanguageClient/AbstractInProcLanguageClient.cs (hunk 2 above)
Comment (CyrusNajmabadi) [reply src/VisualStudio/Core/Def/Implementation/LanguageClient/AbstractInProcLanguageClient.cs]: yeah, i'm reverting this.

**Pull request merged**